src/lib/blueprint.ts

```diff
@@ -127,18 +135,35 @@ interface ObjectProperty extends BaseProperty {
 
 type Method = 'GET' | 'POST' | 'PUT' | 'DELETE' | 'PATCH'
 
-export interface TypesModule {
-  openapi: Openapi
+interface Context {
+  codeSamples: CodeSample[]
 }
 
-export const createBlueprint = ({ openapi }: TypesModule): Blueprint => {
+const TypesModuleSchema = z.object({
+  codeSampleDefinitions: z.array(CodeSampleDefinitionSchema),
+  // TODO: Import and use openapi zod schema here
+  openapi: z.any(),
+})
+
+export type TypesModuleInput = z.input<typeof TypesModuleSchema>
+
+export const createBlueprint = (typesModule: TypesModuleInput): Blueprint => {
+  const { codeSampleDefinitions } = TypesModuleSchema.parse(typesModule)
```

Comment:
Parse the entire input in one step. This will give the best feedback for content authors.